library/list.lisp

```diff
@@ -142,6 +143,12 @@
     "Returns a list containing one element."
     (Cons x Nil))
 
+  (declare singleton? (List :a -> Boolean))
+  (define (singleton? xs)
```

Comment:
can we have singleton for other sequence structures too?

Reply:
For now, I added this to `vector`.